### PR TITLE
Add /sources/:id/(pause|unpause) endpoints for source + application pausing

### PR DESCRIPTION
**Pull request by lindgrenj6**
https://issues.redhat.com/browse/RHCLOUD-14352

**DEPENDS ON:**
- [x] #369 

cc @rvsia 

\# TODO:
- [x] specs

### CODE REVIEW

**Comment by lindgrenj6** [comment]
not sure why the hakiri warning is still there - i updated the gemfile with the new version.

File: Gemfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ gem 'manageiq-messaging',   '~> 1.0.0'
 gem 'manageiq-password',    '~> 0.2', ">= 0.2.1"
 gem 'more_core_extensions', '~> 3.5'
 gem 'pg',                   '~> 1.0', :require => false
-gem 'puma',                 '~> 4.3.5', '>= 4.3.5'
+gem 'puma',                 '~> 4.3.8', '>= 4.3.8'
```
Comment (slemrmartin) [inline Gemfile:18]: it's #405

File: app/controllers/api/v3x1/sources_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module Api
+  module V3x1
+    class SourcesController          < Api::V3x0::SourcesController
+      def pause
+        src = Source.find(params.require(:source_id)).tap { |s| authorize(s) }
+        # the after_discard callback on the Application model handles discarding the source.
+        src.applications.each do |app|
+          app.discard!
+          AvailabilityMessageJob.perform_later("Application.pause", app, Insights::API::Common::Request.current_forwardable)
```
Comment (slemrmartin) [inline app/controllers/api/v3x1/sources_controller.rb:9]: in #369 you changed sending the app to the json format so it should be the same here

File: app/controllers/api/v3x1/sources_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        head 204
+      end
+
+      def unpause
+        src = Source.find(params.require(:source_id)).tap { |s| authorize(s) }
+        # the after_discard callback on the Application model handles undiscarding the source.
+        src.applications.each do |app|
+          app.undiscard!
+          AvailabilityMessageJob.perform_later("Application.unpause", app, Insights::API::Common::Request.current_forwardable)
```
Comment (slemrmartin) [inline app/controllers/api/v3x1/sources_controller.rb:20]: the same, should be `app.to_json`

**Review by slemrmartin — CHANGES_REQUESTED** [review]

File: app/controllers/api/v3x1/sources_controller.rb (hunk 2 above)
Comment (lindgrenj6) [reply app/controllers/api/v3x1/sources_controller.rb]: Sure, updated and marked a dependency on that PR since it will have to go in first.

**Review by slemrmartin — APPROVED** [review]

**Review by lpichler — APPROVED** [review]

**Pull request merged**